src/components/Events.tsx

```diff
@@ -1,44 +1,44 @@
-export const mockEvents = [
-  {
-    id: 1,
-    name: "Vendy",
-    sport: "Basketball",
-  },
-  {
-    id: 2,
-    name: "Lukas",
-    sport: "Football",
-  },
-  {
-    id: 3,
-    name: "Honza",
-    sport: "Tennis",
-  },
-  {
-    id: 4,
-    name: "Vendy",
-    sport: "Box",
-  },
-  {
-    id: 5,
-    name: "Lukas",
-    sport: "Skydiving",
-  },
-  {
-    id: 6,
-    name: "Johana",
-    sport: "Jogging",
-  },
-];
+import { useState, useEffect } from "react";
 
-export const Events: React.FC = () => (
-  <>
-    {mockEvents.map(({ id, sport, name }) => (
-      <article key={id}>
-        <p>{sport}</p>
-        <p>od: {name}</p>
-        <button>Otevřít</button>
-      </article>
-    ))}
-  </>
-);
+interface Event {
+  id: number;
+  sport: string;
+  name: string;
+}
+
+const fetchEventsData = async () => {
+  const response = await fetch("https://sportujspolu-api.onrender.com/events");
+  if (!response.ok) {
+    throw new Error("Failed to fetch data");
+  }
+  const events = await response.json();
+  return events as Event[];
+};
+
+export const Events: React.FC = () => {
+  const [events, setEvents] = useState<Event[]>([]);
+
+  useEffect(() => {
+    (async () => {
+      try {
+        const events = await fetchEventsData();
+        setEvents(events);
+        console.log("events:", events);
+      } catch (error) {
+        console.error(error);
```

Comment:
Even though I understand what you wanted, I appreciate your attention to detail. For now, let's not throw nor catch and we are fine showing 500 error pages in case something bad happens. We should not post anything to the console on production. 

There are multiple approaches to this, which would make sense in the future.

- Do nothing when the resource getting fetched is not important and hide the element f.e. (when data is not present)
- The above has a subpoint. Using error boundary when the section is somewhat important and we want to let the user know this "widget"/section is broken without crashing the whole page
- Show toast/error/warning to the user, when is error results in his interaction (he must know something is fucked)
- Going to error page when some critical resource failed



Ultimately, all errors should be sent to some monitoring service - sentry, datadog, whatever else

Reply:
I will save this for later! 👌